### ISSUE TITLE
Get "TypeError: exception class/object expected" error when update from 3.1.3 to 3.6

### ISSUE THREAD

**Issue by zw963** [sunny/actor#106]
See following backtrace.

```
TypeError: exception class/object expected
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/service_actor-3.6.0/lib/service_actor/result.rb:22:in `raise'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/service_actor-3.6.0/lib/service_actor/result.rb:22:in `fail!'
    /home/common/Stocks/marketbet_crawler/app/services/retrieve_jin10_message.rb:42:in `call'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/service_actor-3.6.0/lib/service_actor/core.rb:48:in `_call'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/service_actor-3.6.0/lib/service_actor/checkable.rb:13:in `_call'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/service_actor-3.6.0/lib/service_actor/defaultable.rb:53:in `_call'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/service_actor-3.6.0/lib/service_actor/failable.rb:33:in `_call'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/service_actor-3.6.0/lib/service_actor/core.rb:16:in `call'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/service_actor-3.6.0/lib/service_actor/core.rb:26:in `result'
    app/routes/jin10/messages.rb:4:in `block (2 levels) in <class:App>'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/hash_paths.rb:120:in `block in hash_paths'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/request.rb:536:in `always'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/hash_paths.rb:120:in `hash_paths'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/hash_routes.rb:296:in `hash_routes'
    /home/common/Stocks/marketbet_crawler/app/app.rb:147:in `block (3 levels) in <class:App>'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/_optimized_matching.rb:140:in `block in _is1'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/request.rb:536:in `always'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/_optimized_matching.rb:140:in `_is1'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/_optimized_matching.rb:25:in `is'
    /home/common/Stocks/marketbet_crawler/app/app.rb:146:in `block (2 levels) in <class:App>'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/request.rb:536:in `always'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/_optimized_matching.rb:124:in `_verb'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/request.rb:104:in `get'
    /home/common/Stocks/marketbet_crawler/app/app.rb:103:in `block in <class:App>'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda.rb:518:in `_roda_run_main_route'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/_before_hook.rb:27:in `_roda_run_main_route'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda.rb:496:in `block in _roda_handle_main_route'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda.rb:494:in `catch'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda.rb:494:in `_roda_handle_main_route'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda/plugins/error_handler.rb:88:in `_roda_handle_main_route'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/roda-3.64.0/lib/roda.rb:380:in `block in base_rack_app_callable'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/rack-test-2.0.2/lib/rack/test.rb:358:in `process_request'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/rack-test-2.0.2/lib/rack/test.rb:165:in `custom_request'
    /home/common/.rvm/gems/ruby-3.2.1@marketbet_crawler/gems/rack-test-2.0.2/lib/rack/test.rb:114:in `get'
    /home/common/.rvm/rubies/ruby-3.2.1/lib/ruby/3.2.0/forwardable.rb:240:in `get'
    /home/common/Stocks/marketbet_crawler/test/routes/jin10/messages_test.rb:6:in `block (2 levels) in <top (required)>'
```

The source code which raise this error is [here](https://github.com/zw963/marketbet_crawler/blob/master/app/services/retrieve_jin10_message.rb)

Thanks.

**Comment by sunny** [comment]
Calling `result.fail!` is not a documented method, but I understand why you’d want to call it. I’ve added a fix in #107 to make it backwards-compatible.

**Comment by sunny** [comment]
Fix released as 3.6.1 🎉
